File: DEFAULT_lib Adapter/app/src/main/java/junseong/k/myapplication/MainActivity.java
package junseong.k.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.CursorAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import android.widget.SimpleCursorAdapter;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.HashMap;

public class MainActivity extends AppCompatActivity implements AdapterView.OnItemClickListener {
    String[] arraydatas;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ListView arrayview=findViewById(R.id.main_listview_array);
        arrayview.setOnItemClickListener(this);
        ListView simpleview=findViewById(R.id.main_listview_simple);
        ListView cursorview=findViewById(R.id.main_listview_cursor);
        arraydatas=getResources().getStringArray(R.array.location);
        ArrayAdapter arrayAdapter=new ArrayAdapter(this,android.R.layout.simple_expandable_list_item_1,arraydatas);
        arrayview.setAdapter(arrayAdapter);

        ArrayList<HashMap<String,String>> simpleDatas=new ArrayList<>();
        DBHelper helper=new DBHelper(this);
        SQLiteDatabase db=helper.getWritableDatabase();
        Cursor cursor=db.rawQuery("select * from tb_data",null);
        while(cursor.moveToNext()){
            HashMap<String,String> map=new HashMap<>();
            map.put("name",cursor.getString(1));
            map.put("content",cursor.getString(2));
            simpleDatas.add(map);
        }
        SimpleAdapter adapter=new SimpleAdapter(this,simpleDatas,android.R.layout.simple_list_item_2,new String[]{"name","content"},
                new int[]{android.R.id.text1,android.R.id.text2});
        simpleview.setAdapter(adapter);

        CursorAdapter cursorAdapter=new SimpleCursorAdapter(this,android.R.layout.simple_list_item_2,cursor,new String[]{"name","content"},new int[]{android.R.id.text1,android.R.id.text2},CursorAdapter.FLAG_REGISTER_CONTENT_OBSERVER);
        cursorview.setAdapter(cursorAdapter);
    }


    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
        Toast.makeText(this, arraydatas[i], Toast.LENGTH_SHORT).show();
    }
}

File: ListView(Kotlin,Simple)/app/src/main/java/junseong/m/listview/MainActivity.kt
package junseong.m.listview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    val data1=Array(20){""}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for(i in 0 until 20)
            data1[i]="문자열${i}"

        val adapter=ArrayAdapter(this,android.R.layout.simple_expandable_list_item_1,data1)
        list1.adapter=adapter
        list1.setOnItemClickListener(listener1)
    }
    val listener1=object:AdapterView.OnItemClickListener{
        override fun onItemClick(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
            when(p0?.id){
                R.id.list1 -> {
                    Toast.makeText(this@MainActivity,"${p2} 번째 클릭",Toast.LENGTH_SHORT).show()
                }
            }
        }
    }
}
